### PR TITLE
Add basic CI using github actions

### LINKED ISSUE #32: setup basic CI
setup basic CI, likely using github actions. We want to run fast checks with this CI, like spelling checks and unit tests.

### PR DESCRIPTION
**Pull request by ffromani**
wire in the minimal CI support with vetting and unit tests

fixes #32

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      with:
+        fetch-depth: 0
+
+    - name: Set up golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.19
+
+    - name: Vet
+      run: go vet ./...
```
Comment (fmuyassarov) [inline .github/workflows/go.yml]: should the Makefile target vet be used instead here?

Comment (ffromani) [reply .github/workflows/go.yml]: it's better indeed, fixing

**Review by fmuyassarov — COMMENTED** [review]
LGTM

**Review by marquiz — COMMENTED** [review]
/lgtm

**Pull request merged**